overlord/ifacestate/ifacestate.go

```diff
@@ -79,10 +113,6 @@ func checkConnectConflicts(st *state.State, change *state.Change, plugSnap, slot
 
 		snapName := snapsup.Name()
 
-		if autoConnectTask != nil && installedSnap == snapName {
```

Comment:
why don't we need this anymore?

Reply:
Good point... I too quickly concluded it's not relevant anymore; Restored and added a comment so I don't forget what is it for. Thanks!